receiver/prometheusreceiver/factory.go

```diff
@@ -49,3 +50,13 @@ func createMetricsReceiver(
 	configWarnings(set.Logger, cfg.(*Config))
 	return newPrometheusReceiver(set, cfg.(*Config), nextConsumer), nil
 }
+
+func configWarnings(logger *zap.Logger, cfg *Config) {
```

Comment:
Did you move it here because this logs a warning message?

Reply:
It is only used in this file, so I think is better to co-locate the code in this case, easier to read (not having to jump files).